kasa/smartdevice.py

```diff
@@ -281,8 +281,11 @@ async def _query_helper(
     @requires_update
     def has_emeter(self) -> bool:
         """Return True if device has an energy meter."""
-        sys_info = self.sys_info
-        features = sys_info["feature"].split(":")
+        return SmartDevice._has_emeter(self.sys_info)
+
+    @staticmethod
+    def _has_emeter(sys_info: Dict[str, Any]) -> bool:
```

Comment:
I personally find doubling of has_emeter as a "public" and "private" method confusing? What about renaming the private one to something like _check_emeter_feature_presence?